### ISSUE TITLE
Error on run

### ISSUE THREAD

**Issue by ryanwaldron** [graylog-labs/cli-dashboard#29]
```
$ graylog-dashboard --server-url http://my.graylogserver.com:1234 --username ***** --password ******
/usr/local/lib/node_modules/graylog-cli-dashboard/graylog-dashboard.js:46
    const {argv} = yargs;
```
Get the above error on run no matter the arguments passed. Installed via NPM on Linux Mint 18.1 using Node v4.2.6

**Comment by STRML** [comment]
That's probably not the whole error, but the issue is the destructuring usage. Use Node >=6.

**Comment by ryanwaldron** [comment]
Running with nodejs v6.10.2 results in:

```
Error: Could not get streams. Error: Bad status code: 404 Not Found [object Object]
    at request.getAsync.then (/usr/local/lib/node_modules/graylog-cli-dashboard/lib/graylog-api.js:98:13)
    at tryCatcher (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/bluebird/js/release/promise.js:693:18)
    at Promise._fulfill (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/bluebird/js/release/promise.js:638:18)
    at Request._callback (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/bluebird/js/release/nodeback.js:42:21)
    at Request.self.callback (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/request/request.js:188:22)
    at emitTwo (events.js:106:13)
    at Request.emit (events.js:191:7)
    at Request.<anonymous> (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/request/request.js:1171:10)
    at emitOne (events.js:96:13)
    at Request.emit (events.js:188:7)
    at IncomingMessage.<anonymous> (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/request/request.js:1091:12)
    at IncomingMessage.g (events.js:291:16)
    at makeRequest.then.catch (/usr/local/lib/node_modules/graylog-cli-dashboard/lib/graylog-api.js:65:13)
    at tryCatcher (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/bluebird/js/release/promise.js:689:18)
    at Async._drainQueue (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/bluebird/js/release/async.js:133:16)
    at Async._drainQueues (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/bluebird/js/release/async.js:143:10)
    at Immediate.Async.drainQueues (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/bluebird/js/release/async.js:17:14)
    at runCallback (timers.js:666:20)
    at tryOnImmediate (timers.js:639:5)
    at processImmediate [as _immediateCallback] (timers.js:611:5)
```

Verified that the serverURL, username and password are correct.

**Comment by ryanwaldron** [comment]
Using Graylog 2.1.3+040d371 if that has an impact.

**Comment by AndyBehr** [comment]
I have the same issue. 
graylog-cli installed on macOS
node v6.10.0
Graylog 2.2.0+d9681cb

**Comment by rozling** [comment]
Having the exact same issues.

`rest_listen_uri` is http://127.0.0.1:12900/api/

 first got the 'yargs' error with node 4.2.6:
```
graylog-dashboard --stream-title "Test" --server-url http://127.0.0.1:12900/api
/usr/local/lib/node_modules/graylog-cli-dashboard/graylog-dashboard.js:46
    const {argv} = yargs;
          ^

SyntaxError: Unexpected token {
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:374:25)
    at Object.Module._extensions..js (module.js:417:10)
    at Module.load (module.js:344:32)
    at Function.Module._load (module.js:301:12)
    at Function.Module.runMain (module.js:442:10)
    at startup (node.js:136:18)
    at node.js:966:3
```

Upgraded node to latest version v6.11.0 & got following error:

```
graylog-dashboard --stream-title "Test" --server-url http://127.0.0.1:12900/api
? Please input username: admin
? Please input password: *****
Error: Could not get streams. Error: Bad status code: 404 Not Found [object Object]
    at request.getAsync.then (/usr/local/lib/node_modules/graylog-cli-dashboard/lib/graylog-api.js:98:13)
    at tryCatcher (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/bluebird/js/release/promise.js:693:18)
    at Promise._fulfill (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/bluebird/js/release/promise.js:638:18)
    at Request._callback (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/bluebird/js/release/nodeback.js:42:21)
    at Request.self.callback (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/request/request.js:188:22)
    at emitTwo (events.js:106:13)
    at Request.emit (events.js:191:7)
    at Request.<anonymous> (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/request/request.js:1171:10)
    at emitOne (events.js:96:13)
    at Request.emit (events.js:188:7)
    at IncomingMessage.<anonymous> (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/request/request.js:1091:12)
    at IncomingMessage.g (events.js:292:16)
    at makeRequest.then.catch (/usr/local/lib/node_modules/graylog-cli-dashboard/lib/graylog-api.js:65:13)
    at tryCatcher (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/bluebird/js/release/promise.js:689:18)
    at Async._drainQueue (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/bluebird/js/release/async.js:133:16)
    at Async._drainQueues (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/bluebird/js/release/async.js:143:10)
    at Immediate.Async.drainQueues (/usr/local/lib/node_modules/graylog-cli-dashboard/node_modules/bluebird/js/release/async.js:17:14)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
```

output of `http get http://127.0.0.1:12900/api`:
```
HTTP/1.1 307 Temporary Redirect
Content-Length: 0
Date: Sun, 25 Jun 2017 10:50:37 GMT
Location: http://127.0.0.1:12900/
X-Graylog-Node-ID: 16b3d1e6-20ee-452f-b3ab-4f37276266f6
X-Runtime-Microseconds: 177
```

**Comment by okelet** [comment]
Same here.

**Comment by erpel** [comment]
When specifying URLs on the command line, that ended in /api/, I ran into the problem, that the script would issue http GET requests to a path like `/api/:12900/streams`. To rectify this incorrect URL completion (inserting the port into the path part of the URL), I created a pull request, that addresses part of the issue. #30 

This solution is far from perfect but should be minimally invasive. A better option might be, to expand the command line arguments to contain a separate host and port option or to do full URL parsing to avoid appending a colon and port number to the URL in inappropriate places.
